### ISSUE TITLE
Add a GUIs tab

### ISSUE THREAD

**Issue by pascal-ballet** [jeancharles-roger/matrixstudio#21]
It could be interesting to add a GUIs (or Views) tab before the Matrices tab to combine Matrices in one.
Thus we can create graphic user interfaces in a simple way.
This is how can it be done:
- With a button "Add a matrix", a rectangle is placed, moved and resized by the user. Then the matrix displayed by the rectangle is selected thanks to a combo box. Several rectangles can be added to create a multi-view of matrices.

A question is: do we need only one GUI (View) or more?
Probably more than one to create fully functional software.
It induces the question: how to switch between GUIs (Views)?
- By adding clickable rectangle? (i.e: OnClik, GoTo GUI2)
- Will we have to add conditions/actions on global parameters? (ex. OnClik, Cdt:state==1, Act: state++, GoTo GUI2)
- other?

**Comment by jeancharles-roger** [comment]
Adding a GUI creation tool to matrix studio is quite a challenge. I think that we'll quickly need more. I'll advise to integrate matrices inside other UI (JavaFX SceneBuilder for instance). 

To achieve this, we'll need #22 to be figured out

**Comment by pascal-ballet** [comment]
Yes, you're right. I added a new boolean attribute in MatrixStudio.java called Embedded (false by default). If its value is true, the GUI is not opened, so it can be used into any new java project.
Here is how it can be used with JavaFX (note this is a first step that must be improved):

import matrixstudio.kernel.Simulator;
import matrixstudio.model.Matrix;
import matrixstudio.ui.MatrixStudio;

public class FXMLDocumentController implements Initializable {
    MatrixStudio ms;
    Simulator s;
    Matrix mat;
    
    @Override
    public void initialize(URL url, ResourceBundle rb) {
        // Set Matrix Studio Embedded
        ms = new  MatrixStudio();
        ms.SetEmbedded(true);
        ms.open();
        // Load the Matrix Studio Simulation of your choice
        ms.loadModel( "yourPath/yourSimulation.mss");
        // Get the Simulator to control the OpenCL program
        s = ms.getSimulator();
        // Compile the OpenCL program
        s.compileKernelCode();
        // Execute the Matrix Studio Simulation
        s.executeStep(0, 0, 0);
        // Get the results from the first matrix
        mat = s.getModel().getMatrix(0);
        // etc
    }
